examples/encodeRadioDetails/encodeRadioDetails.ino

```diff
@@ -0,0 +1,225 @@
+            (static_cast<bool>(autoAck & _BV(ENAA_P4)) + 48),
+            (static_cast<bool>(autoAck & _BV(ENAA_P3)) + 48),
+            (static_cast<bool>(autoAck & _BV(ENAA_P2)) + 48),
+            (static_cast<bool>(autoAck & _BV(ENAA_P1)) + 48),
+            (static_cast<bool>(autoAck & _BV(ENAA_P0)) + 48)
+        }
+     25 bool (read_register(NRF_CONFIG) & _BV(PRIM_RX))
+     26 uint8_t tx_address[5]
+        {
+            TX_ADDR
+        }
+
+        //indicate whether pipes are open or closed after first address
+     31 uint8_t pipe_forty_bit_address_array[5]
+        {
+            static_cast<uint8_t>(RX_ADDR_P0 + 0),
+            static_cast<uint8_t>(RX_ADDR_P0 + 1)
+        }
+     41 uint8_t pipe_eight_bit_register_array[4]
+        {
+            static_cast<uint8_t>(RX_ADDR_P0 + 2),
+            static_cast<uint8_t>(RX_ADDR_P0 + 3),
+            static_cast<uint8_t>(RX_ADDR_P0 + 4),
+            static_cast<uint8_t>(RX_ADDR_P0 + 5)
+        }
+*/
+
+/*
+  https://www.rapidtables.com/convert/number/binary-to-decimal.html
+  Enter in your binary number and this will convert it to decimal (base10) for you
+  a uint16_t has 16 bits, a uint8_t has 8 bits, a boolean value is packed into one bit
+  in the bit array.  Trailing zeros are padding
+ */
+
+void loop() {
+
+  if (role) {
+    // This device is a TX node
+
+    unsigned long start_timer = micros();                    // start the timer
+    bool report = radio.write(&payload, sizeof(payload));      // transmit & save the report
+    unsigned long end_timer = micros();                      // end the timer
+
+    if (report) {
+      Serial.print(F("Transmission successful! "));          // payload was delivered
+      Serial.print(F("Time to transmit = "));
+      Serial.print(end_timer - start_timer);                 // print the timer result
+      Serial.println(F(" us. Sent: "));
+      for (int i = 0; i < 10; i++) {  //print out each member of payload[] in binary
+        Serial.println(payload[i], BIN);  //you can manually decode by following the packing order of encodeRadioDetails
+      }
+    } else {
+      Serial.println(F("Transmission failed or timed out")); // payload was not delivered
+    }
+
+    // to make this example readable in the serial monitor
+    delay(60000);  // slow transmissions down by 1 minute
+
+  } else {
+    // This device is a RX node
+
+    uint8_t pipe;
+    if (radio.available(&pipe)) {             // is there a payload? get the pipe number that recieved it
+      uint8_t bytes = radio.getPayloadSize(); // get the size of the payload
+      radio.read(&payload, bytes);            // fetch payload from FIFO
+      Serial.print(F("Received "));
+      Serial.print(bytes);                    // print the size of the payload
+      Serial.print(F(" bytes on pipe "));
+      Serial.print(pipe);                     // print the pipe number
+      Serial.println(F(": "));
+      for (int i = 0; i < 10; i++) {
```

Comment:
Would it be more transparent to use `bytes` var instead of hard coding this loop to `10` iterations?

Reply:
I'm not sure what you mean.

Reply:
```cpp
/*
  For this example, we'll be using a payload containing
  radio details encoded with RF24::encodeRadioDetails()
  it is sent by the sender and decoded by the receiver
  using decodeRadioDetails(), you can manually decode
  your encoded details by printing out the payload value
  in binary by or blinking a led for each bit in the
  payload
*/
uint32_t payload[10] = {0};
```
Looks like all payloads in this example are 10 bytes long.

```cpp
  // save on transmission time by setting the radio to only transmit the
  // number of bytes we need to transmit
  radio.setPayloadSize(sizeof(payload)); // float datatype occupies 4 bytes
```
The comment here should now describe the payload as a byte array (&not a float value).

```cpp
    uint8_t pipe;
    if (radio.available(&pipe)) {             // is there a payload? get the pipe number that recieved it
      uint8_t bytes = radio.getPayloadSize(); // get the size of the payload
```
Will this `bytes` var always be equal to 10? Seems like it should.

If so, `bytes` should probably be used to iterate over the payload when printing out the binary representation. By saying "transparent", I mean that users will be able to read/understand that the entire payload is iterated (& not just the first arbitrary 10 bytes).

Reply:
I see, you are correct.  I think I should also change the serial output to reflect that it's a "binary representation of the payload" or something.

Reply:
yeah, don't want to let them think they chose the wrong baudrate for `Serial.begin()`

Reply:
I just realized that each element in the payload is a `uint32_t`... 10 uint32_t is not 10 bytes. I think it comes out to be 40 actual bytes which is greater than the max payload size?! Am I calculating that wrong?

Reply:
it is 40 bytes, we can make it less than 40 by getting rid of a couple of members or by splitting the transmit in the loop, I forgot we aren't using network!

Reply:
The radio's Mac payload length is 32. This is a hardware limit not a lib limit.

Reply:
If using the networking layers, then the max payload length is 24 without message fragmentation (a feature specific to RF24Network). This is due to needed header info. Fragmentation is disabled for ATTiny devices by default (enabled for all others).

Reply:
I split the payload in the example, each half is 24 bytes